src/libraries/System.Private.CoreLib/src/System/Threading/LazyInitializer.cs

```diff
@@ -48,8 +48,12 @@ public static class LazyInitializer
         /// if an object was not used and to then dispose of the object appropriately.
         /// </para>
         /// </remarks>
-        public static T EnsureInitialized<T>([NotNull] ref T? target) where T : class =>
-            Volatile.Read(ref target) ?? EnsureInitializedCore(ref target);
+        public static T EnsureInitialized<T>([NotNull] ref T? target) where T : class
+        {
+            T value = Volatile.Read(ref target) ?? EnsureInitializedCore(ref target);
+            NullabilityHelpers.SuppressNonNullAssignmentWarning(ref target);
```

Comment:
Why is this necessary?

Reply:
Ah, I see.  We've annotated Volatile.Read's return as being `[NotNullIfNotNull("location")]`, which will tell the compiler that the result is non-null if location is non-null, but the compiler can't infer from that that location is non-null if the result is non-null, and we lack a `[NotNullIfResultNotNull]` attribute.  Ugh.

I'd rather just address this with a `!` on the `ref target`.

Reply:
The way we'd address this would be to put a bang on the first _target_, not the second _target_:

```cs
return Volatile.Read(ref target!) ?? EnsureInitializedCore(ref target);
```

That's technically a lie, since the bang is in the wrong spot. But since it's really just an internal implementation detail of this method I suppose that's ok and we shouldn't worry too much about it?

Reply:
> I suppose that's ok and we shouldn't worry too much about it?

Yup.  Thanks.